x-pack/platform/plugins/shared/fleet/server/tasks/upgrade_agentless_deployment.ts

```diff
@@ -0,0 +1,276 @@
+    this.logger = logFactory.get(this.taskId);
+
+    taskManager.registerTaskDefinitions({
+      [UPGRADE_AGENTLESS_DEPLOYMENTS_TASK_TYPE]: {
+        title: TITLE,
+        timeout: TIMEOUT,
+        createTaskRunner: ({ taskInstance }: { taskInstance: ConcreteTaskInstance }) => {
+          return {
+            run: async () => {
+              return this.runTask(taskInstance, core);
+            },
+            cancel: async () => {
+              this.abortController.abort(`${TITLE} timed out`);
+            },
+          };
+        },
+      },
+    });
+  }
+
+  public start = async ({ taskManager }: UpgradeAgentlessDeploymentsTaskStartContract) => {
+    if (!taskManager) {
+      this.logger.error(`${LOGGER_SUBJECT} Missing required service during start`);
+    }
+
+    this.startedTaskRunner = true;
+    this.logger.info(`${LOGGER_SUBJECT} Started with interval of [${INTERVAL}]`);
+
+    try {
+      await taskManager.ensureScheduled({
+        id: this.taskId,
+        taskType: UPGRADE_AGENTLESS_DEPLOYMENTS_TASK_TYPE,
+        schedule: {
+          interval: INTERVAL,
+        },
+        state: {},
+        params: { version: UPGRADE_AGENT_DEPLOYMENTS_TASK_VERSION },
+      });
+    } catch (e) {
+      this.logger.error(`Error scheduling task ${LOGGER_SUBJECT}, error: ${e.message}`, e);
+    }
+  };
+
+  private get taskId(): string {
+    return `${UPGRADE_AGENTLESS_DEPLOYMENTS_TASK_TYPE}:${UPGRADE_AGENT_DEPLOYMENTS_TASK_VERSION}`;
+  }
+
+  private endRun(msg: string = '') {
+    this.logger.info(`${LOGGER_SUBJECT} runTask ended${msg ? ': ' + msg : ''}`);
+  }
+
+  private processInBatches = async (
+    {
+      agentlessPolicies,
+      agents,
+    }: {
+      agentlessPolicies: AgentPolicy[];
+      agents: Agent[];
+    },
+    batchSize: number,
+    processFunction: (agentPolicy: AgentPolicy, agentlessAgent: Agent) => void
+  ) => {
+    for (let i = 0; i < agentlessPolicies.length; i += batchSize) {
+      const currentAgentPolicyBatch = agentlessPolicies.slice(i, i + batchSize);
+
+      await Promise.allSettled(
+        currentAgentPolicyBatch.map((agentPolicy) => {
+          const agentlessAgent = agents.find((agent) => agent.policy_id === agentPolicy.id);
+
+          if (!agentlessAgent) {
+            this.endRun('No active online agentless agent found');
+            return;
+          }
+
+          this.logger.info(
+            `${LOGGER_SUBJECT} processing agentless agent ${JSON.stringify(agentlessAgent.agent)}`
+          );
+
+          return processFunction(agentPolicy, agentlessAgent);
+        })
+      );
+    }
+  };
+
+  private processUpgradeAgentlessDeployments = async (
+    esClient: ElasticsearchClient,
+    soClient: SavedObjectsClient
+  ) => {
+    const SAVED_OBJECT_TYPE = 'fleet-agent-policies';
+
+    const policiesKuery = `${SAVED_OBJECT_TYPE}.supports_agentless: true`;
+
+    try {
+      const agentPolicyFetcher = await agentPolicyService.fetchAllAgentPolicies(soClient, {
```

Comment:
We probably want to update that method to support a `spaceId: *` and use a saved object without space extension to be able to retrieve all policies across all space, as we can now enable fleet to be space aware.